### PR TITLE
main_handle_no_tracked_remote_branch

### PR DESCRIPTION
**Pull request by Simontwice**
# Description
todo

# Checklist
Answer each question with "yes", and provide an explanation if the answer isn't what is expected.

1. Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - Answer:
2. Are all variables set with logical defaults that are backward compatible?
   - Answer:
3. Can you provide a link to a Neptune experiment verifying the functionality of this PR?
   - Answer:
4. Have you successfully executed the linter and tests?
   - Answer:
5. Are all functions concise and don't require splitting?
   - Answer:
6. Is there documentation for arguments and complex logic?
   - Answer:
7. Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - Answer:
8. Are all functions placed in appropriate files?
   - Answer:
9. Is the PR name meaningful and descriptive?
   - Answer:
10. Is PR description provided, or unnecessary?
    - Answer:
  
Other questions:

11. Have you not made changes to the requirements or anything related to the Singularity image? If you have, did you generate and upload a new image to the clusters?
    - Answer:
12. Is the change significant enough to notify all individuals working with this repository?
    - Answer:
13. Is there a need for a second reviewer?
    - Answer:

### CODE REVIEW

File: lizrd/support/code_versioning.py [hunk 1]
```diff
@@ -41,6 +41,14 @@ def version_code(self):
...
             self.original_branch_commit_hash = self.repo.head.object.hexsha
 
+            # check if the branch is tracking any remote branch at origin
+            assert (
+                self.repo.active_branch.tracking_branch() is not None
+            ), "Branch is not tracking any remote branch. Aborting..."
+            assert (
+                self.repo.active_branch.tracking_branch().remote_name == "origin"
+            ), "Branch is not ta branch located at origin. Aborting..."
+
```
Comment (crewtool) [inline lizrd/support/code_versioning.py:51]: One line maybe?
```suggestion
            # check if the branch is tracking any remote branch at origin
            assert (
                self.repo.active_branch.tracking_branch() is not None and self.repo.active_branch.tracking_branch().remote_name == "origin" 
            ), "Branch is not tracking any remote branch. Aborting..."
```

**Review by crewtool — COMMENTED** [review]
Also why?

**Comment by Simontwice** [comment]
> Also why?

cannot check whether you have unpushed commits, if the branch does not track a remote branch, as it happens by comparing the two.

**Review by crewtool — APPROVED** [review]
W sumie pewnie przydałby się w przyszłości dać możliwość braku origin

**Pull request merged**